app/controllers/api/requests_controller.rb

```diff
@@ -1,5 +1,35 @@
 module Api
   class RequestsController < BaseController
+    include Subcollections::RequestTasks
+    include Subcollections::Tasks
+
+    def create_resource(type, _id, data)
+      assert_id_not_specified(data, type)
+
+      request_type = data.delete("request_type")
+      validate_request_type(request_type)
+
+      if data["src_id"].blank? && data["src_ids"].blank?
+        raise BadRequestError, "Must specify a resource src_id or src_ids"
+      end
+
+      requester = Hash(data.delete("requester"))
+      user = fetch_requester_user(requester)
+      auto_approve = fetch_auto_approve(requester)
+
+      request_type.constantize.create_request(data.symbolize_keys, user, auto_approve)
+    end
+
+    def edit_resource(type, id = nil, data = {})
```

Comment:
What is `type`?  I'm assuming STI type, but not sure why we would need that if we have an ID.